etlman/projects/views.py

```diff
@@ -200,7 +200,17 @@ def new_step_step2(request, project_id, step_id=None):
             "name": session_pipeline["pipeline-name"] + "_script",
             "script": "def main():\n\t...\nif __name__ == '__main__':\n\tmain()",
         }
-        form_step = StepForm(instance=step, initial=initial)
+        if not step:
+            form_step = StepForm(instance=step, initial=initial)
+        else:
+            form_step = StepForm(
+                instance=step,
+                initial={
```

Comment:
`initial` would [never be used](https://docs.djangoproject.com/en/4.1/ref/forms/api/#initial-form-values) here, if I understand correctly. There might be something else going on, or perhaps the bug did not exist to begin with?

Reply:
I tested again in `main` and in this branch and the bug does exist 🤔